src/context/TodoContext.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { ErrorList } from '../types/ErrorList';
+import { Status } from '../types/Status';
+import { wait } from '../utils/fetchClient';
+import { deleteTodo, getTodos } from '../api/todos';
+
+export type TodoContext = {
```

Comment:
We already have a variable named `TodosContext`.

Currently, it's a bit confusing to differentiate between `TodoContext` and `TodosContext`.

Let's rename this type.

```suggestion
export type TodosContextType = {
```